.github/workflows/ci.yml

```diff
@@ -0,0 +1,755 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+#### Global environment variables
+env:
+  DYNAMODB_HOSTNAME: dynamodb
+  MYSQL_HOSTNAME: mysql
+  MYSQL_DATABASE: testdb
+  MYSQL_USERNAME: root
+  MYSQL_PASSWORD: Password123
+  DISABLE_TESTCONTAINERS: true
+  AWS_ACCESS_KEY_ID: dummy_key
+  AWS_SECRET_ACCESS_KEY: dummy_secret
+  AWS_DEFAULT_REGION: us-west-2
+  AWS_REGION: us-west-2
+  GO111MODULE: "on"
+
+
+jobs:
+  #### Java
+  java-secure-memory:
+    name: Build Java Secure Memory
+    runs-on: ubuntu-latest
+    container:
+      image: adoptopenjdk/maven-openjdk11
+      options: --ulimit core=-1 --ulimit memlock=-1:-1
+    steps:
+    - name: Checkout the repository
+      uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+    - name: Cache Java packages
+      uses: actions/cache@1a9e2138d905efd099035b49d8b7a3888c653ca8
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - name: Build
+      run: |
+        cd java/secure-memory
+        ./scripts/clean.sh
+        ./scripts/build.sh
+    - name: Test
+      run: |
+        cd java/secure-memory
+        ./scripts/test.sh
+
+  release-java-secure-memory:
+    if: contains( github.ref, 'release' )
```

Comment:
When exactly does this make the release happen? When a branch is pushed? When a PR is opened? When an update is made to a PR? When a PR is merged? Could this potentially lead to duplicate deployments?

Reply:
Whenever a push happens on a branch with `release` in the name. I've updated it to `release-` now. It will not lead to duplicate deployments because the `release_prod` scripts check if the tag for that release already exists. If it does, the process is skipped.